stdlib/testing/kapacitor/combine_join.flux

```diff
@@ -0,0 +1,58 @@
+package testdata_test
+
+import "testing"
+
+option now = () => (2030-01-01T00:00:00Z)
+
+inData = "
+#datatype,string,long,string,dateTime:RFC3339,double,string
+#group,false,false,true,false,false,true
+#default,_result,,,,,
+,result,table,_measurement,_time,_value,user
+,,0,memory,2018-12-19T22:13:30Z,1,user1
+,,0,memory,2018-12-19T22:13:40Z,5,user1
+,,0,memory,2018-12-19T22:13:50Z,3,user1
+,,0,memory,2018-12-19T22:14:00Z,6,user1
+,,0,memory,2018-12-19T22:14:10Z,6,user1
+,,0,memory,2018-12-19T22:14:20Z,3,user1
+,,1,memory,2018-12-19T22:13:30Z,6,user2
+,,1,memory,2018-12-19T22:13:40Z,7,user2
+,,1,memory,2018-12-19T22:13:50Z,3,user2
+,,1,memory,2018-12-19T22:14:00Z,4,user2
+,,1,memory,2018-12-19T22:14:10Z,9,user2
+,,1,memory,2018-12-19T22:14:20Z,8,user2
+"
+
+outData = "
+#datatype,string,long,string,dateTime:RFC3339,double,double,string,string
+#group,false,false,true,false,false,false,true,false
+#default,_result,,,,,,,
+,result,table,_measurement,_time,_value_left,_value_right,user_left,user_right
+,,0,memory,2018-12-19T22:13:30Z,1,6,user1,user2
+,,0,memory,2018-12-19T22:13:40Z,5,7,user1,user2
+,,0,memory,2018-12-19T22:13:50Z,3,3,user1,user2
+,,0,memory,2018-12-19T22:14:00Z,6,4,user1,user2
+,,0,memory,2018-12-19T22:14:10Z,6,9,user1,user2
+,,0,memory,2018-12-19T22:14:20Z,3,8,user1,user2
+"
+
+t_combine_join = (table=<-) => {
+    left = table
+		|> range(start: 2018-12-19T22:13:30Z, stop: 2018-12-19T22:14:30Z)
+		|> drop(columns: ["_start", "_stop"])
+		|> filter(fn: (r) =>
+			(r.user == "user1"))
+		|> group(columns: ["user"])
+
+    right = table
+		|> range(start: 2018-12-19T22:13:30Z, stop: 2018-12-19T22:14:30Z)
+		|> drop(columns: ["_start", "_stop"])
+		|> filter(fn: (r) =>
+			(r.user == "user2"))
+		|> group(columns: ["_measurement"])
+
+    return join(tables: {left: left, right: right}, on: ["_time", "_measurement"])
```

Comment:
was just looking at this, I think I oversimplified this somewhat.   `combine`  operates on a single stream, it doesn't join 2 streams.  It also has an interesting feature,  `tolerance`  that we need to see if we can support (maybe/probably not).    

I think that `pivot`  using `_time, _measurement`  as a rowKey is closer in nature to combine.